### DISCUSSION TITLE
Multiple Captain Services / Instances to scale API

### DISCUSSION THREAD

**Discussion by KellerKev** [caprover/caprover#1967] [Q&A]
Currently I am interfacing with caprover only via API through a scheduler (Python Celery). 

Sometimes I would like to send multiple requests to Caprover to deploy and build and image and update the config for antoher app, but then I get the 429 error that an operation is in progress. 

So I am trying to find ways to scale the Caprover API as it seems the API is blocking and can only take one request at a time. 

Can I make the API non-blocking somehow or can I scale Captaian to run multiple instances via Docker Swarm? 

Wehen I tried to increase instances I get the error that port 3000 is already in use.. 

So that did not work for me so far.

Or can I scale out caprover to another server/node in order to parallelize requests?

Any help appreciated.

**Comment by githubsaturn** [comment]
Scaling CapRover isn't possible since data is atomically shared in `/captain/data` directory and it will cause data corruption if you have more than one instance running. 

CapRover isn't designed for this use case, and it takes a major rewrite to queue the requests sent via the API. Lots of edge cases (what if request failed due to any error, should proceed or not etc...)

Instead, you should add the business logic your your scheduler, something like:

```
// api.createNewApp
// api.getAllApps
// check if the new app is created
// api.addNewDomain
// check if the domain is added
// .... 
```
